src/lib/message_filters.py

```diff
@@ -124,3 +125,47 @@ def filter_noise(messages, message_key, noise_fn):
         log.info(f"Filtered out messages identified as noise. "
                  f"Returning {len(filtered)}/{len(messages)} messages.")
         return filtered
+
+    @staticmethod
+    def filter_katikati_sms_survey_messages(messages, time_keys):
+        """
+        Filters out a list of messages received during Katikati sms survey(s) broadcast period.
+
+        :param messages: List of message objects to filter.
+        :type messages: list of TracedData
+        :param time_keys: Keys in each TracedData object that contain the time the message was sent.
+                          Each TracedData should have exactly one match for each key.
+                          The values must be strings in ISO 8601 format.
+        :type time_keys: set of str
+        :return: Filtered list.
+        :rtype: list of TracedData
+        """
+        # De-duplicate time_keys
+        assert isinstance(time_keys, set)
```

Comment:
I know this is copied from a function above, but perhaps time_keys = set(time_keys) would be better

Reply:
updated thanks.